crowbar_framework/app/models/service_object.rb

```diff
@@ -1238,15 +1348,21 @@ def apply_role(role, inst, in_queue)
             nodes_in_batch << node_name unless nodes_in_batch.include?(node_name)
           end
         end
-      end
+      end # roles.each
+
       @logger.debug "nodes_in_batch #{nodes_in_batch.inspect}"
       run_order << nodes_in_batch unless nodes_in_batch.empty?
       @logger.debug "run_order #{run_order.inspect}"
     end
 
+    # Part III: Update run lists of nodes to reflect new deployment. I.e. write through
+    # the deployment schedule in pending node actions into run lists.
     @logger.debug "Clean the run_lists for #{pending_node_actions.inspect}"
+
     admin_nodes = []
+
     pending_node_actions.each do |node_name, lists|
+      # FIXME: Do we constantly need to filter out nodes that drop out during deployment?
```

Comment:
I think we've done it so far for the nodes in the new deployment, but not for the nodes in the old deployment (ie, nodes for which the proposal will be deactivated).